src/screens/WalletScreen/index.tsx

```diff
@@ -36,7 +36,8 @@ function WalletScreen() {
   const { params } = useRoute<RouteProp<RootStackParamList, 'WalletScreen'>>();
   const { setParams, getState: dangerouslyGetState, getParent: dangerouslyGetParent } = useNavigation();
   const removeFirst = useRemoveFirst();
-  const [initialized, setInitialized] = useState(!!params?.initialized);
+  const initializedWallet = useRef(false);
+  const initializingWallet = useRef(false);
```

Comment:
Is there a need for both of these controls for essentially controlling the same state, can we use an enum based approach?
```
enum WalletLoadingStates {
   IDLE = 0
   INITIALIZING = 1
   INITIALIZED = 2
}

const walletState = useRef(State.IDLE);
...

walletState.current = WalletLoadingStates.INITIALIZING;
// @ts-expect-error messed up initializeWallet types
await initializeWallet(null, null, null, !params?.emptyWallet);
walletState.current = WalletLoadingStates.INITIALIZED;
```

Reply:
Yeah that's cleaner